### DISCUSSION TITLE
Add object with petrubed trajectory

### DISCUSSION THREAD

**Discussion by pl-phan** [Stellarium/stellarium#2443] [Q&A]
I want to preview the 2029 Apophis flyby, when it passes at 32000 km from the Earth.
Is there a way to setup an Apophis object with correct ephemeris ?

All I managed to do was setup an object with the *instantaneous* heliocentric orbital parameters of Apophis at its closest Earth distance, resulting in a vaguely correct flyby preview (from Paris).
But it keeps constant orbital parameters, it does not take into account Earth's gravity which should make significant changes to a flyby preview.
Any idea if this is possible ?

**Comment by gzotti** [comment]
It is currently not possible.

**Reply by pl-phan** [reply]
thank you for your answer

**Reply by pl-phan** [reply]
@gzotti But I can get the ephemeris data from an external source (JPL Horizons), then is there no way to use it ?
For example it seems that Mars has a special treatment (`coord_func=mars_special`), possibly because of perturbations, are there resources somewhere to know how this works ?

**Reply by alex-w** [reply]
See an example: https://github.com/Stellarium/stellarium/pull/1942

**Reply by gzotti** [reply]
I said it is currently not possible. Minor bodies run on unperturbed Keplerian orbits. If YOU extend the program code by implementing a coord_func=apophis_special, we will gladly add it. It is not on my current agenda, but who knows how this program will evolve until ~2028?
